solutions/swb-reference/src/services/sshKeyService.ts

```diff
@@ -127,8 +128,49 @@ export default class SshKeyService implements SshKeyPlugin {
    * @returns a {@link CreateSshKeyResponse}
    */
   public async createSshKey(request: CreateSshKeyRequest): Promise<CreateSshKeyResponse> {
-    // TODO: implement
-    throw new Error('Method not implemented.');
+    const { projectId, userId } = request;
+    const sshKeyId = this._getSshKeyId(userId, projectId);
+
+    // get envMgmtRoleArn and externalId from project record
+    const { envMgmtRoleArn, externalId } = await this._getEnvMgmtRoleArnAndExternalIdFromProject(projectId);
+
+    // get EC2 client
+    const { ec2 } = await this._getEc2ClientsForHostingAccount(
+      envMgmtRoleArn,
+      'Create',
+      externalId,
+      this._aws
+    );
+
+    // create key
+    try {
+      const createKeyPairResponse = await ec2.createKeyPair({
+        KeyFormat: KeyFormat.pem,
+        KeyName: sshKeyId,
+        KeyType: KeyType.rsa,
+        TagSpecifications: [
+          {
+            ResourceType: 'key-pair',
+            Tags: [
+              {
+                Key: 'user',
+                Value: userId
+              },
+              {
+                Key: 'project',
+                Value: projectId
+              }
+            ]
+          }
+        ]
+      });
+      return { projectId, privateKey: createKeyPairResponse.KeyMaterial!, sshKeyId, owner: userId };
+    } catch (e) {
+      if (e.Code === 'InvalidKeyPair.Duplicate') {
+        throw new DuplicateKeyError(e.message);
```

Comment:
it might be capitalized: e.Message from the example: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/errors-overview.html#api-error-response

Reply:
Hmm. I tried this. Weirdly enough, the error message only propagated to the response when I used lower case `e.message` and I got the generic message when I try to pass `e.Message`. I'm going to stick to lower case for now.